File: outdoor-saas-fe/src/services/planBarrierService.ts
/**
 * 方案道闸明细服务
 */
import request from './api';
import type { PageResult } from '../types/query';

export interface PlanBarrier {
  id: number;
  planId: number;
  barrierGateId: number;
  planCommunityId: number;
  releaseDateBegin?: string;
  releaseDateEnd?: string;
  releaseStatus?: number;
  plan?: {
    id: number;
    planNo: string;
    planName: string;
  };
  barrierGate?: {
    id: number;
    gateNo: string;
    deviceNo: string;
  };
  planCommunity?: {
    id: number;
    communityId: number;
  };
}

export interface PlanBarrierQueryParam {
  pageNum?: number;
  pageSize?: number;
  planId?: number;
  barrierGateId?: number;
  releaseStatus?: number;
}

export const planBarrierService = {
  // 获取所有方案道闸明细
  getAll: () => request<PlanBarrier[]>('/plan-barrier/list'),

  // 根据ID获取
  getById: (id: number) => request<PlanBarrier>(`/plan-barrier/${id}`),

  // 分页查询（带过滤参数）
  filterPage: (params: PlanBarrierQueryParam) =>
    request<PageResult<PlanBarrier>>('/plan-barrier/filter/page', {
      method: 'POST',
      body: JSON.stringify(params),
    }),

  // 根据方案ID获取
  getByPlan: (planId: number) =>
    request<PlanBarrier[]>(`/plan-barrier/plan/${planId}`),

  // 根据道闸ID获取
  getByBarrier: (barrierGateId: number) =>
    request<PlanBarrier[]>(`/plan-barrier/barrier/${barrierGateId}`),

  // 根据方案社区ID获取
  getByPlanCommunity: (planCommunityId: number) =>
    request<PlanBarrier[]>(`/plan-barrier/plan-community/${planCommunityId}`),

  // 根据发布状态获取
  getByStatus: (releaseStatus: number) =>
    request<PlanBarrier[]>(`/plan-barrier/status/${releaseStatus}`),

  // 创建
  create: (data: Partial<PlanBarrier>) =>
    request<PlanBarrier>('/plan-barrier', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  // 批量创建
  createBatch: (data: Partial<PlanBarrier>[]) =>
    request<PlanBarrier[]>('/plan-barrier/batch', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  // 更新
  update: (data: Partial<PlanBarrier>) =>
    request<PlanBarrier>('/plan-barrier', {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  // 删除
  delete: (id: number) =>
    request<void>(`/plan-barrier/${id}`, { method: 'DELETE' }),

  // 批量删除
  deleteBatch: (ids: number[]) =>
    request<void>('/plan-barrier/batch', {
      method: 'DELETE',
      body: JSON.stringify(ids),
    }),
};

export default planBarrierService;

File: outdoor-saas-be/src/main/java/com/touhuwai/service/PlanBarrierService.java
package com.touhuwai.service;

import com.touhuwai.common.PageResult;
import com.touhuwai.entity.PlanBarrier;

import java.util.List;

/**
 * 方案道闸明细服务接口
 */
public interface PlanBarrierService {
    
    /**
     * 根据ID查询方案道闸明细
     * @param id 明细ID
     * @return 方案道闸明细信息
     */
    PlanBarrier getById(Integer id);
    
    /**
     * 查询所有方案道闸明细
     * @return 方案道闸明细列表
     */
    List<PlanBarrier> getAll();
    
    /**
     * 分页查询方案道闸明细列表
     * @param planBarrier 查询条件
     * @param pageNum 页码
     * @param pageSize 每页大小
     * @return 分页结果
     */
    PageResult<PlanBarrier> getPage(PlanBarrier planBarrier, Integer pageNum, Integer pageSize);
    
    /**
     * 根据方案ID查询明细列表
     * @param planId 方案ID
     * @return 方案道闸明细列表
     */
    List<PlanBarrier> getByPlanId(Integer planId);
    
    /**
     * 根据道闸ID查询明细列表
     * @param barrierGateId 道闸ID
     * @return 方案道闸明细列表
     */
    List<PlanBarrier> getByBarrierGateId(Integer barrierGateId);
    
    /**
     * 根据方案社区ID查询明细列表
     * @param planCommunityId 方案社区ID
     * @return 方案道闸明细列表
     */
    List<PlanBarrier> getByPlanCommunityId(Integer planCommunityId);
    
    /**
     * 根据发布状态查询明细列表
     * @param releaseStatus 发布状态
     * @return 方案道闸明细列表
     */
    List<PlanBarrier> getByReleaseStatus(Integer releaseStatus);
    
    /**
     * 新增方案道闸明细
     * @param planBarrier 方案道闸明细信息
     * @return 影响的行数
     */
    int add(PlanBarrier planBarrier);
    
    /**
     * 批量新增方案道闸明细
     * @param list 方案道闸明细列表
     * @return 影响的行数
     */
    int batchAdd(List<PlanBarrier> list);
    
    /**
     * 更新方案道闸明细
     * @param planBarrier 方案道闸明细信息
     * @return 影响的行数
     */
    int update(PlanBarrier planBarrier);
    
    /**
     * 根据ID删除方案道闸明细
     * @param id 明细ID
     * @return 影响的行数
     */
    int delete(Integer id);
    
    /**
     * 批量删除方案道闸明细
     * @param ids ID列表
     * @return 影响的行数
     */
    int batchDelete(List<Integer> ids);
}
